### PR TITLE
Added parameters to tools/format.sh (Windows)

### PR DESCRIPTION
**Pull request by meastp**
This adds parameters for skipping the sed operations and/or the operations that require installed software (clang-format, cmake-format ...).

The usage scenario is that @g-easy 's Docker-container is not useable in Windows Docker because the sed operations fail. However, the workaround is easy:
- Install Git for Windows - this gives you a bash shell. Run format.sh from here, skipping the operations requiring installed software
- Use @g-easy 's Docker container for the operations requiring installed software.

I have tested this, and it works very well :)

### CODE REVIEW

File: tools/format.sh [hunk 1]
```diff
@@ -18,37 +18,69 @@ if [[ ! -e tools/format.sh ]]; then
...
+
+usage()
+{
+    echo "usage: format.sh [--disable-sed-check] [--disable-tools-check]"
+}
+
+sed_check=y
+tools_check=y
+
+while [ "$1" != "" ]; do
```
Comment (g-easy) [inline tools/format.sh]: `[[ ... ]]` for consistency with the rest of the file.

File: tools/format.sh [hunk 2]
```diff
@@ -57,3 +89,9 @@ if [[ ! -z "$CHANGED" ]]; then
 else
   echo "No changes."
 fi
+
+if [ "$tools_check" == "$tools_check" ]; then
```
Comment (g-easy) [inline tools/format.sh]: This doesn't look right. Also it breaks Travis.

Comment (meastp) [reply tools/format.sh]: ups - should have been $sed_check == $tools_check :)

**Comment by g-easy** [comment]
It passes on Travis now. @meastp are you happy with my edits / the state of this PR in general?

**Comment by meastp** [comment]
Thank you for the fixes, @g-easy - LGTM :)

**Pull request merged**